rfc/networked-dot-storage.md

```diff
@@ -0,0 +1,174 @@
+The first step on the path for W3S is indeed to build other implementations of w3up. Local.storage is our first experiment. It would be nice to produce a cloudflare only implementation. The architecture proposed in [Core Node Architecture](https://w3sat.notion.site/Filecoin-SPs-in-the-W3S-Network-16a9b66e7ef3425bbee390c6481bd752?pvs=4) is a kind of aspirational target -- an ultraportable implementation that could run in most server environments, be ultra-sandboxed, provide great update mechanisms, and support rapid iteration.
+
+We should constinue to explore these options.
+
+### Router.storage
+
+We need an implementation of W3up that simply routes to other instances of W3Up. Building this implies several tasks:
+- deciding who gets each piece of content
+- how to manage UCAN delegations across relays
+
+And eventually:
+- how to allow nodes to join and how to decide when to kick them out
+
+But first, we should probably just hardcode a list of nodes.
+
+*** But isn't this a point of centralization if everyone uses it? ***
+
+Absolutely. It's important that key coordination functions in the network like router.storage eventually operate through distributed consensus and produce verifiable results.
+
+### Read Pipeline that finds content from many write pipelines
+
+A retrieval service (i.e. Freeway) should not be tied to a specific instance of w3up. This implies some form of global content discovery. [W3 IPNI Indexing RFC](https://github.com/web3-storage/RFC/blob/main/rfc/ipni-w3c.md) is a major step in this direction. 
+
+Eventually we may need some kind of regional caching.
+
+### Repair service for both W3UP and Filecoin
+
+Repair service means a service that creates permaneant reliable storage from a series of time-limited storage deals with providers who might fail.
+
+It's a vital missing component today on Filecoin AND a missing component of W3S as soon as their is more than one instance. We've already encountered this problem indirectly as we've thought about deal renewals and the reality that deal terms on Filecoin do not coincide in time with users joining and leaving W3UP.
+
+It needs to exist for Filecoin AND it needs to exist for W3S.
+
+*** But isn't this a point of centralization if everyone uses it? ***
+
+Absolutely. It's important that key coordination functions in the network like a repair service eventually operate through distributed consensus and produce verifiable results.
+
+### Shared w3filecoin pipeline
+
+W3Filecoin is an excellent service for aggregating data on to Filecoin. At the same time, running it is NOT easy. While the basic aggregation functions are simple, making Filecoin deals is not. Multiple instances of W3UP should be able to share an instance of W3Filecoin. 
+
+*** But isn't this a point of centralization if everyone uses it? ***
+
+Absolutely. It's important that key coordination functions in the network like w3-filecoin eventually operate through distributed consensus and produce verifiable results.
```

Comment:
ditto, same, "I used a blockchain to get your data to the blockchain" decentralized inception?! 

but tbh of all the "exit to consensus" questions this one makes the most sense to me? the filecoin pipeline is already so async that a blockchain outage slowing it down feels fine, and having all of the intermediary steps show up on chain somewhere seems pretty reasonable.

Reply:
I mean this service would likely run as a Filecoin L2, so really it's just a sub routine :P